lib/sensors/temperature.cpp

```diff
@@ -15,20 +32,19 @@ Temperature::~Temperature()
 {
 }
 
-std::optional<std::int16_t> Temperature::read()
+std::optional<core::Result> Temperature::isValueReady()
 {
   const auto status_check_result = i2c_->readByte(kTemperatureDefaultAddress, kStatus);
   if (!status_check_result) {
     logger_.log(
       core::LogLevel::kFatal, "Failed to read temperature sensor status at channel %d", channel_);
     return std::nullopt;
   }
-  // TODOLater: change this from a fatal error to a debug and re-prompt read (with ROS)
   if (status_check_result.value() == kBusy) {
-    logger_.log(core::LogLevel::kFatal,
+    logger_.log(core::LogLevel::kWarn,
                 "Failed to read, temperature sensor is not ready to be read at channel %d",
                 channel_);
-    return std::nullopt;
+    return core::Result::kFailure;
   }
   if (status_check_result.value() == kTemperatureOverUpperLimit) {
```

Comment:
This feels like it's beyond the scope of the function?

Reply:
i understand that, however the alternative would be to do another i/o operation in `read` itself (i.e. checking the same register but for fatal errors), which I think is worse

Reply:
Surely you could just implement a helper function or similar instead?

Reply:
since this entire function deals with just the status register, I think splitting it would be redundant. I do think that maybe its name could be better? Maybe like `checkStatus` ?

Reply:
Yeah this sounds good. Just something that implies the functionality more intuitively.